wordpress-mobile/WordPress-iOS
# Posts & Pages: Allow setting a posts page back to a regular page

## Description
- Adds an option for setting a posts page back to a regular page, to match behavior on Calypso
- Fixes actions for a homepage (can't trash a homepage or move it to drafts)

Fix homepage actions | Set post page as reg page
-- | --
<img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/c148c400-e834-43a8-a077-73bb2757442d" width=200> | <img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/1987575b-c12f-401b-a1ca-8304e77705c3" width=200>

## How to test
- Go to the pages list screen
- Verify that a homepage doesn't have the "move to draft" or "move to trash" options
- Verify that a posts page has the "set as regular page" option
- Verify that "set as regular page" works as expected

## Regression Notes
1. Potential unintended areas of impact
pages context menu

2. What I did to test those areas of impact (or what existing automated tests I relied on)
PageMenuViewModelTests

3. What automated tests I added (or what prevented me from doing so)
see above

PR submission checklist:


UI Changes testing checklist:
